### PR TITLE
Fix #10929 Graphic UI selection glitch with selected text on large screens

### LINKED ISSUE #10929: Graphic UI selection glitch with selected text on large screens
Hi,
I have a large screen (5120x1440) and I display the Pharo Window on all the available surface.
This is great but I have a funny bug :D when I select a text (from a source-code, or anything) the selection rectangle render is not visible.

See some screenshots after.

My Pharo desktop : 
image

And the selection bug in differents cases :
DirtySelection

On this gif we can see that there is "an invisible line", at the left off it the render is visible and at the right the render is cutted. May be this is the same problem for the vertical axe.

This is a problem when my selection is completely at the left of this line, I don't see the selection :S 
InvisibleSelection

This invisible line appear to be at 4095 pixels of the beginning of my screen (with a measure app), is the render clipped outside 4096 pixels ?

My config : 
Pharo 10.0.0
Build information: Pharo-10.0.0+build.399.sha.c4426366e83c26b8a002ef0680abe3a89a0b1a99 (64 Bit)
On Windows 10 Pro 64bits.

### PR DESCRIPTION
**Pull request by labordep**
My reflexion in the concerned thread here : #10929

### CODE REVIEW

**Comment by MarcusDenker** [comment]
- [CI] Windows testrunner has problems #12763

File: src/FormCanvas-Core/BalloonEngine.class.st [hunk 1]
```diff
@@ -426,7 +426,11 @@ BalloonEngine >> flush [
 BalloonEngine >> initialize [
 	| w |
 	super initialize.
-	w := Display width > 2048 ifTrue: [ 4096 ] ifFalse: [ 2048 ].
+	
+	"Get the display width to initialize width of the bitmap, if the width is zero or negative, use a standard value as 2048"
+	w := Display width.
+	w <= 0 ifTrue:[w := 2048].
```
Comment (demarey) [inline src/FormCanvas-Core/BalloonEngine.class.st]: 2048 is magic number.
It should be nice to extract it in a method like `defaultBitmapWidth`.
And also, renaming `w`to `width` would be nice too.

Comment (labordep) [reply src/FormCanvas-Core/BalloonEngine.class.st]: thanks @demarey, I will do that :)

File: src/FormCanvas-Core/BalloonEngine.class.st [hunk 2]
```diff
@@ -50,6 +50,13 @@ BalloonEngine class >> debug: aBoolean [
 	Debug := aBoolean
 ]
 
+{ #category : #accessing }
+BalloonEngine class >> defaultBitmapWidth [
+	"Return default bitmap width"
```
Comment (demarey) [inline src/FormCanvas-Core/BalloonEngine.class.st]: The comment is a paraphrase of the selector. I would remove it.

**Review by demarey — APPROVED** [review]
Thank you for the changes and the contribution.
I suggest a last change but is not mandatory for the inclusion of this PR.

File: src/FormCanvas-Core/BalloonEngine.class.st (hunk 2 above)
Comment (jecisc) [reply src/FormCanvas-Core/BalloonEngine.class.st]: I agree with Christophe

Comment (labordep) [reply src/FormCanvas-Core/BalloonEngine.class.st]: Ok ;-)
I will remove that.

**Review by jecisc — APPROVED** [review]

**Pull request merged**